src/libraries/Microsoft.PowerFx.Core/Public/Types/UserProvidedTypeDefinitions/UserDefinedRecordType.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core
+{
+    internal class UserDefinedRecordType : RecordType
```

Comment:
What does this mean?

Reply:
It's a record type that's defined by the user in a Yaml definition. As it can reference other type names and could be recursive, it's a great candidate for a derived RecordType.